web/app.js

```diff
@@ -1297,39 +1301,109 @@ var PDFViewerApplication = {
     }
   },
 
-  bindWindowEvents: function pdfViewBindWindowEvents() {
-    var eventBus = this.eventBus;
-
-    window.addEventListener('wheel', webViewerWheel);
-    window.addEventListener('click', webViewerClick);
-    window.addEventListener('keydown', webViewerKeyDown);
+  bindWindowEvents() {
+    let eventBus = this.eventBus;
 
-    window.addEventListener('resize', function windowResize() {
+    this._boundEvents.windowResize = () => {
       eventBus.dispatch('resize');
-    });
-    window.addEventListener('hashchange', function windowHashChange() {
+    };
+    this._boundEvents.windowHashChange = () => {
       eventBus.dispatch('hashchange', {
         hash: document.location.hash.substring(1),
       });
-    });
-    window.addEventListener('beforeprint', function windowBeforePrint() {
+    };
+    this._boundEvents.windowBeforePrint = () => {
       eventBus.dispatch('beforeprint');
-    });
-    window.addEventListener('afterprint', function windowAfterPrint() {
+    };
+    this._boundEvents.windowAfterPrint = () => {
       eventBus.dispatch('afterprint');
-    });
-    if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
-      window.addEventListener('change', function windowChange(evt) {
-        var files = evt.target.files;
-        if (!files || files.length === 0) {
-          return;
-        }
-        eventBus.dispatch('fileinputchange', {
-          fileInput: evt.target,
-        });
+    };
+    this._boundEvents.windowChange = (evt) => {
```

Comment:
https://github.com/mozilla/pdf.js/pull/8525#discussion_r121949328 still appears to be unresolved in the latest version of the patch.